packages/svelte/src/compiler/phases/2-analyze/visitors/shared/component.js

```diff
@@ -15,6 +16,70 @@ import { mark_subtree_dynamic } from './fragment.js';
  * @param {Context} context
  */
 export function visit_component(node, context) {
+	node.metadata.path = [...context.path];
+
+	// link this node to all the snippets that it could render, so that we can prune CSS correctly
+	node.metadata.snippets = new Set();
+
+	let resolved = true;
+
+	for (const attribute of node.attributes) {
+		/** @type {Expression | undefined} */
+		let expression;
+
+		if (attribute.type === 'SpreadAttribute' || attribute.type === 'BindDirective') {
+			resolved = false;
+			continue;
+		}
+
+		if (attribute.type !== 'Attribute' || attribute.value === true) {
+			continue;
+		}
+
+		if (Array.isArray(attribute.value)) {
+			if (attribute.value.length === 1 && attribute.value[0].type === 'ExpressionTag') {
+				expression = attribute.value[0].expression;
+			}
+		} else {
+			expression = attribute.value.expression;
+		}
+
+		if (!expression) continue;
+
+		if (expression.type === 'Identifier') {
+			const binding = context.state.scope.get(expression.name);
+
+			if (
+				binding &&
+				binding.declaration_kind !== 'import' &&
+				binding.kind !== 'prop' &&
+				binding.kind !== 'rest_prop' &&
+				binding.kind !== 'bindable_prop' &&
+				binding.initial?.type !== 'SnippetBlock'
```

Comment:
I don't understand why we need all the additional "not this" checks, and why in only this specific case resolve is set to false - could use a comment